pkg/controlplane/deleted_kinds.go

```diff
@@ -82,8 +85,21 @@ func newResourceExpirationEvaluator(currentVersion apimachineryversion.Info) (*r
 	return ret, nil
 }
 
-func (e *resourceExpirationEvaluator) shouldServe(resourceServingInfo rest.Storage) bool {
-	versionedPtr := resourceServingInfo.New()
+func (e *resourceExpirationEvaluator) shouldServe(gv schema.GroupVersion, versioner runtime.ObjectVersioner, resourceServingInfo rest.Storage) bool {
+	internalPtr := resourceServingInfo.New()
```

Comment:
why Ptr?

Reply:
why was this a versioned object before?

Reply:
> why was this a versioned object before?

the name was wrong.  It was always internal .  :(